paltax/source_models.py

```diff
@@ -437,8 +461,12 @@ def __init__(self, cosmos_path: str, catalog_weights: jnp.ndarray):
             catalog_weights: Weights for the sources in the catalog. Do not
                 need to be normalized.
         """
-        # Save the cosmos image path.
-        super().__init__(cosmos_path=cosmos_path)
+
+        # Saves the cosmos path, chunk size, and opens the hdf5 file
+        super().__init__(cosmos_path=cosmos_path, images_per_chunk=images_per_chunk)
+
+        # Loads the catalog weights corresponding to the input parameter
+        catalog_weights = self.hdf5_file[parameter + '_weights']
```

Comment:
You should save the parameter value, not the catalog_weights. Technically this would also work, but I think it would be clearer if we accessed the hdf5 file in the modify_cosmology_params call.